comet/redis_async_locks.py

```diff
@@ -76,11 +76,15 @@ def __init__(self, redis, name):
     async def close(self):
         """Clean up the database entries for the lock.
 
-        This will acquire the lock before removing it.
+        This will acquire the lock before removing it. Raises a LockError if the lock
+        can't get acquired.
         """
-        await self.acquire()
+        r = await self.acquire(no_block=True)
```

Comment:
> Should this just be replaced with an unconditional deletion of the list in redis? It's simpler, more robust, and I think has the right properties.

Do you mean just closing the lock without locking it?

Reply:
Yeah. Scratch most of that objection.

I guess one issue could be that if the lock is closed on one worker, but it is locked elsewhere, it will fail, but the other worker could subsequently release it. I think that means there is just a general subtlety in closing the lock, if you want to guarantee a clean shutdown you need to actually blocking acquire.

One thing that might help is changing the release to use an `LPUSHX`. That way if the lock has been closed (i.e. the list removed), no subsequent `.release()` can recreate it.

Reply:
I am not sure how to put your comments together. At the moment the lock can't be closed while it is acquired elsewhere. Are you referring to an alternative implementation of `close()`?